athena-oracle/src/main/java/com/amazonaws/athena/connectors/oracle/OracleJdbcConnectionFactory.java

```diff
@@ -60,22 +70,52 @@ public Connection getConnection(final JdbcCredentialProvider jdbcCredentialProvi
     {
         try {
             final String derivedJdbcString;
-            if (null != jdbcCredentialProvider) {
-                Matcher secretMatcher = SECRET_NAME_PATTERN.matcher(databaseConnectionConfig.getJdbcConnectionString());
-                final String secretReplacement = String.format("%s/%s", jdbcCredentialProvider.getCredential().getUser(), jdbcCredentialProvider.getCredential().getPassword());
-                derivedJdbcString = secretMatcher.replaceAll(Matcher.quoteReplacement(secretReplacement));
+            final Map<String, String> envVariables = System.getenv();
+
+            if (OracleConstants.YES.equalsIgnoreCase(envVariables.get("ssl"))) {
+                derivedJdbcString = databaseConnectionConfig.getJdbcConnectionString();
+                Pattern connStringPattern = Pattern.compile("jdbc:oracle:thin:@\\((?i)description=\\(address=\\(protocol=tcps\\)\\(host=[a-zA-Z0-9-.]+\\)" +
+                        "\\(port=([1-9][0-9]{0,3}|[1-5][0-9]{4}|6[0-4][0-9]{3}|65[0-4][0-9]{2}|655[0-2][0-9]|6553[0-5])\\)\\)" +
+                        "\\(connect_data=\\(sid=[a-zA-Z_]+\\)\\)\\(security=\\(ssl_server_cert_dn=\\\"[=a-zA-Z,0-9-.,]+\\\"\\)\\)\\)");
+                Matcher connStringMatcher = connStringPattern.matcher(derivedJdbcString);
+                if (!connStringMatcher.matches()) {
+                    throw new RuntimeException("Invalid connection string to establish connection over SSL, Please check.");
+                }
+
+                JdbcCredential credential = getCredentialsFromSecretsManager(envVariables.get("secret_name"));
+                this.jdbcProperties.put("user", credential.getUser());
+                this.jdbcProperties.put("password", credential.getPassword());
```

Comment:
Is there a reason that we can't just reuse the existing mechanism for getting the username/password secret?

If the reason is because of the uri validation, I would suggest to validate only the part after the secret.

Like just split the derivedJdbcString using "@" and then only run the last part part through the pattern matcher.

The pattern must also be modified accordingly to validate the part without the username and password of course.

Then you can continue to make use of the `jdbcCredentialProvider`.

Reply:
yes Henry,

we used to provide secret name along with connection string.
I see below SSL based connection string from [oracle documentation](https://www.oracle.com/technetwork/database/enterprise-edition/wp-oracle-jdbc-thin-ssl-130128.pdf)  which does not have credentials as part of url.
`String url = "jdbc:oracle:thin:@(DESCRIPTION=(ADDRESS=(PROTOCOL=tcps)(HOST=servername)(PORT=2484))(CONNECT_DATA=(SERVICE_NAME=servicename)))";`

As the earlier logic has failed in SSL scenario, I have added utility method which will fetch credentials from aws secret manager.

Today I have tried sample java program which uses SSL based connection string along with credentials as part of the url. It worked fine, now I've reused the existing mechanism as suggested.

Reply:
Great thanks for checking and updating the patch.